Don't swallow errors during prerender build

### Command

build

### Description

We migrated to newer Angular version (`@angular/cli@21.0.4`), and we got a lot of build errors with `prerender: true`.

The errors at least for route prerendering are currently very hard to pinpoint, because the error can happen in any component/module/etc that the route uses and there's no error stack.

The errors look like this:

```
✘ [ERROR] An error occurred while prerendering route '/en/trait-db/browse'.

Cannot read properties of undefined (reading 'trait-search-ignore')
```

The errors are generated here:

https://github.com/angular/angular-cli/blob/main/packages/angular/build/src/utils/server-rendering/prerender.ts#L263

This line swallows the error stack, because there's an error message almost always present and the newly created error doesn't include the original stack.

I tried adding `console.log(error)` above that line, and then I could see the error stack.

Error stacks are crucial. Could error management improved during building?

### Describe the solution you'd like

Log error stack during build with prerender

### Describe alternatives you've considered

_No response_

--- a/packages/angular/build/src/utils/server-rendering/prerender.ts
+++ b/packages/angular/build/src/utils/server-rendering/prerender.ts
@@ -116,8 +116,12 @@ export async function prerenderPages(
     sourcemap,
     outputMode,
   ).catch((err) => {
+    assertIsError(err);
+
     return {
-      errors: [`An error occurred while extracting routes.\n\n${err.message ?? err.stack ?? err}`],
+      errors: [
+        `An error occurred while extracting routes.\n\n${err.stack ?? err.message ?? err.code ?? err}`,
+      ],
       serializedRouteTree: [],
       appShellRoute: undefined,
     };
@@ -265,8 +269,9 @@ async function renderPages(
           }
         })
         .catch((err) => {
+          assertIsError(err);
           errors.push(
-            `An error occurred while prerendering route '${route}'.\n\n${err.message ?? err.stack ?? err.code ?? err}`,
+            `An error occurred while prerendering route '${route}'.\n\n${err.stack ?? err.message ?? err.code ?? err}`,
           );
           void renderWorker.destroy();
         });
@@ -371,7 +376,7 @@ async function getAllRoutes(
 
     return {
       errors: [
-        `An error occurred while extracting routes.\n\n${err.message ?? err.stack ?? err.code ?? err}`,
+        `An error occurred while extracting routes.\n\n${err.stack ?? err.message ?? err.code ?? err}`,
       ],
       serializedRouteTree: [],
     };

